File: client/src/types/item.ts
export interface Item {
  id: string;
  title: string;
  image?: string | null;
  type: string;
  buyPrice: number;
  rentPrice: number;
  isRent: boolean;
  isSale: boolean;
  description?: string | null;
  flatNumber?: string;
  houseNumber: string;
  street: string;
  city: string;
  country: string;
  params?: {
    hasKitchen?: boolean;
    hasWorkZone?: boolean;
    hasParking?: boolean;
    hasGym?: boolean;
    hasWifi?: boolean;
    hasTV?: boolean;
    hasIron?: boolean;
    hasPool?: boolean;
    hasElevator?: boolean;
  },
  host?: string;
  roomCount: number;
  createdAt?: string;
  updatedAt?: string;
}

export const itemTypeOptions = [
  {
    label: 'Дом',
    value: 'house',
  },
  {
    label: 'Квартира',
    value: 'flat',
  },
  {
    label: 'Комната',
    value: 'room',
  },
  {
    label: 'Апартаменты',
    value: 'aparts',
  },
  {
    label: 'Вилла',
    value: 'villa',
  },
];


export enum ITEM_PARAMS {
  hasKitchen = 'Кухня',
  hasWorkZone = 'Рабочая зона',
  hasGym = 'Спортзал',
  hasParking = 'Парковка',
  hasPool = 'Бассейн',
  hasWifi = 'Wi-Fi',
  hasTV = 'Телевизор',
  hasIron = 'Утюг',
}

export interface ItemBody extends Omit<Item, 'id'> {}

File: api/src/main/java/api/api/Item/Item.java
package api.api.Item;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonProperty;

@Document(collection = "items")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Item {
    @Id
    private String id;
    private String title;
    private String image;
    private String type;
    private Number rentPrice;
    private Number buyPrice;
    @JsonProperty("isRent")
    private boolean isRent;
    @JsonProperty("isSale")
    private boolean isSale;
    private Number roomCount;
    private String description;
    private String flatNumber;
    private String houseNumber;
    private String street;
    private String city;
    private String country;
    private Object params;
    private String host;
    @CreatedDate
    private LocalDateTime createdAt;
    @LastModifiedDate
    private LocalDateTime modifiedAt;
}
